docs/modules/ROOT/pages/dapp.adoc

```diff
@@ -0,0 +1,337 @@
+= Building a Dapp with Solidity Hot Loader
+
+== Introduction
+
+In this guide, we will build a simple Dapp using the Solidity Hot Loader. We will create a Counter Dapp, which displays the value of a Counter that can be increased. We will update the code in the smart contract to show the power of the Solidity Hot Loader to automatically update the Dapp.
+
+== Setup
+
+We will use the https://create-react-app.dev/[`create-react-app`] package to bootstrap a React application.
+
+First we need a local blockchain to develop on. We use `ganache-cli`, so go ahead and install (if you haven’t already) and then run a local blockchain using `ganache-cli`.
+
+[source,console]
+----
+$ npm install -g ganache-cli
+$ ganache-cli --deterministic
+----
+
+Next we need a project directory to develop our project in. In a separate terminal create the project directory and create `package.json` with `npm init`.
+
+[source,console]
+----
+$ mkdir myproject && cd myproject
+$ npm init -y
+----
+
+We use the OpenZeppelin CLI for compiling and deploying our contract, so go ahead and install *OpenZeppelin CLI*.
+
+[source,console]
+----
+$ npm install @openzeppelin/cli
+----
+
+Next, initialize the project with OpenZeppelin. This will create directories and config for our project.
+
+[source,console]
+----
+$ npx openzeppelin init
+? Welcome to the OpenZeppelin SDK! Choose a name for your project myproject
+? Initial project version 1.0.0
+Project initialized. Write a new contract in the contracts folder and run 'openzeppelin create' to deploy it.
+----
+
+== Counter contract
+
+Our Counter smart contract consists of an integer `value` and a function to increase it by one.
+
+In the `contracts` directory save `Counter.sol` with the following contents:
```

Comment:
```suggestion
```
Skip in favor of comments on the code - I find the setup becomes too verbose otherwise